### ISSUE TITLE
Function with name 'GroupArrayIntersect' does not exist

### ISSUE THREAD

**Issue by pufit** [ClickHouse/ClickHouse#67582]
**Describe the unexpected behaviour**
```sql
WITH
    (
        SELECT [1]
    ) AS array,
    (
        SELECT groupArrayIntersect(array)
    ) AS intersect
SELECT intersect
```

```
Received exception from server (version 24.7.1):
Code: 46. DB::Exception: Received from localhost:9000. DB::Exception: Function with name 'GroupArrayIntersect' does not exist. In scope SELECT GroupArrayIntersect(__getScalar('18303850344086609523_3217598707281124672')) AS `groupArrayIntersect(array)` FROM system.one. Maybe you meant: ['groupArrayIntersect','groupArrayInsertAt']. (UNKNOWN_FUNCTION)
```

Reproduces on maser

**Expected behavior**
```
┌─groupArrayIntersect(array)─┐
│ [1]                        │
└────────────────────────────┘
```

**Error message and/or stacktrace**
https://pastila.nl/?000a241f/554cd9d33bae29c25e4f2e5c7b4c717c#06lW2Fy+2zGGsrNdCsg5ng==

**Comment by pufit** [comment]
cc @yariks5s

**Comment by Algunenano** [comment]
Already fixed in [24.7](https://fiddle.clickhouse.com/3c9ea122-fefd-4da8-a44b-a50656648c88) -> https://github.com/ClickHouse/ClickHouse/pull/66928/files#diff-978664b7b641bf6b250eb72a816f1b875cd95b2e21c5882bcefb380bf4338863R212
